src/components/connect.js

```diff
@@ -413,21 +414,18 @@ function connect(mapPropsToRequestsToProps, defaults, options) {
     RefetchConnect.displayName = `Refetch.connect(${getDisplayName(WrappedComponent)})`
     RefetchConnect.WrappedComponent = WrappedComponent
 
-    if (dependsOnContext && WrappedComponent.contextTypes) {
-      RefetchConnect.contextTypes = WrappedComponent.contextTypes
-    }
-
     if (process.env.NODE_ENV !== 'production') {
-      RefetchConnect.prototype.componentWillUpdate = function componentWillUpdate() {
+      RefetchConnect.prototype.componentDidUpdate = (previous => function componentDidUpdate() {
         if (this.version === version) {
+          previous.apply(this, arguments)
           return
         }
 
         // We are hot reloading!
         this.version = version
         this.clearAllRefreshTimeouts()
         this.refetchDataFromProps()
-      }
+      })(RefetchConnect.prototype.componentDidUpdate)
```

Comment:
@tkvw I know you've moved on from this PR (thanks for getting it started BTW!), but I've been working on reviving it to hopefully fix this once and for all. I have a question about the change to this section. Do you happen to remember why you needed to call `previous.apply(this, arguments)` in here?  It seems like if `this.version === version`, this should not be needed, but I could very easily be missing something.